### ISSUE TITLE
Replayer should supply block specific context to transactions

### ISSUE THREAD

**Issue by frankbraun** [near/evm-bully#3]
@djsatok brought up the issue on Discord that smart contracts can use `block.timestamp` and if that doesn't match a replayed transaction will fail. 

It's even worse then that, all of the following could potentially be used:

- `block.coinbase`  
- `block.timestamp`   
- `block.number`
- `block.difficulty`
- `block.gaslimit`  
- `hash = block.blockHash(blockNumber)`   (hash of the specific block, only valid for the 256 most recent blocks, excluding the current one)

See [EVM Opcodes](https://www.ethervm.io/).

After discussion with @artob, taking into consideration that the EVM will now mostly be implemented as a smart contract, we agreed that the best approach would be to extend the [near-web3-provider](https://github.com/near/near-web3-provider) API with a something like `Eth.send_raw_transaction_with_context`. Similar to `Eth.send_raw_transaction`, but also passing in the necessary block context to execute the transaction within a *test* contract deployment of the EVM.

That way, in combination with issue #2, we should be able to replay all transactions of an Ethereum testnet to a NEAR EVM test deployment.

**Comment by djsatok** [comment]
@frankbraun , one important thing would be to guard `Eth.send_raw_transaction_with_context`, and most importantly, corresponding EVM contract method with a `#feature[bully]`, which will be used only for stress-testing and will never be deployed in production environment.

**Comment by frankbraun** [comment]
@djsatok: Yes, absolutely. I think we should stress test on betanet that way, maybe on testnet (to be determined), but the corresponding functionality (in `nearcore` and `near-web3-provider`) should never be activated on mainnet.

**Comment by frankbraun** [comment]
The block context is now prepared by the replayer.

**Comment by artob** [comment]
As described in https://github.com/aurora-is-near/aurora-engine/issues/1#issuecomment-797495269 and https://github.com/aurora-is-near/aurora-engine/issues/2#issuecomment-797492075, the EVM contract is now ready to begin accepting `begin_chain` and `begin_block` calls. The implementation of those functions is a work in progress, but they can be called already, unblocking the Bully to proceed calling them.
